lib/asset-manifest-generator.js

```diff
@@ -141,4 +138,44 @@ function stringOrRegexMatch(matchers, value) {
   return false;
 }
 
+/**
+ * @typedef {Object} BundleInfo
+ * @property {boolean} isValid - true if we have a valid new bundle name or a bundle and asset name
+ * @property {boolean} isNewBundle - true if this entry is just a bundle name
+ * @property {string} bundleName
+ * @property {string=} assetName
+ * @property {string=} assetType - asset extension
+ *
+ * @param {string} entry
+ * @returns {BundleInfo}
+ */
+function bundleInfoFromAssetEntry(entry) {
```

Comment:
I really like splitting this out into a stand alone function, which should make testing various combinations much easier. Can you export this function privately, and add some unit tests around expected inputs and outputs?

Reply:
Good idea! Adding tests gave me an opportunity to clean up some quirks that carried over from the original implementation.